src/segwit.rs

```diff
@@ -0,0 +1,227 @@
+// SPDX-License-Identifier: MIT
+
+//! Segregated Witness API - enables typical usage for encoding and decoding segwit addresses.
+//!
+//! The bips contain some complexity, this module should allow you to create modern Bitcoin
+//! addresses, and parse existing addresses from the Bitcoin blockchain, correctly and easily.
```

Comment:
In 047a29e89017a5a4a4f4ccdbdec9bbad69c7f11c:

`bips` should be capitalized as `BIPs`. `,` should be a period.

`parse existing addresses from the blockchain` is not correct; addresses don't directly appear on the blockchain anywhere.

Reply:
Oh wow, I had totally gotten it into my head that the HRP was somehow part of on chain data, I needed this comment of yours, thanks. I now feel much better about our handling of the HRP (not enforcing the bc/bt thing).